tests/DocBlock/TypeExpressionTest.php

```diff
@@ -234,6 +234,17 @@ public static function provideGetTypesCases(): iterable
         yield ['array{a: int, b: int, c: int, d: int, e: int, f: int, g: int, h: int, i: int, j: int, with-dash: int}'];
 
         yield ['array{a: int, b: int, c: int, d: int, e: int, f: int, g: int, h: int, i: int, j: int, k: int, l: int, with-dash: int}'];
+
+        yield [sprintf(
```

Comment:
the "backtrack limit" was a problem when the type was not matched, so we should add also some large types that are invalid, like in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7843#discussion_r1493524794

Reply:
Added, rebased to have 2 commits (instead of 8) when in 1st both new test cases - valid and invalid are failing.